### PR TITLE
Increase resourceSyncer code coverage

### PR DESCRIPTION
**Pull request by tpantelis**
See individual commits for details.

Code coverage increased from 67% to 93.6%,

### CODE REVIEW

**Review by dfarrell07 — APPROVED** [review]
Nice!

File: pkg/syncer/resource_syncer.go [hunk 1]
```diff
@@ -61,16 +61,18 @@ const (
 )
 
 func (d SyncDirection) String() string {
+	s := ""
```
Comment (skitt) [inline pkg/syncer/resource_syncer.go]: The behaviour is different,

```suggestion
	s := "unknown"
```

would preserve the coverage improvement and keep the existing behaviour. I know it’s rather theoretical, but a `SyncDirection` can take any `int` value...

File: pkg/syncer/resource_syncer.go [hunk 2]
```diff
@@ -82,16 +84,18 @@ const (
 )
 
 func (o Operation) String() string {
+	s := ""
```
Comment (skitt) [inline pkg/syncer/resource_syncer.go]:
```suggestion
	s := "unknown"
```

File: pkg/syncer/resource_syncer.go [hunk 3]
```diff
@@ -322,89 +326,99 @@ func (r *resourceSyncer) ListResources() []runtime.Object {
...
 		}
-	}()
+
+		obj := resourceUtil.MustToUnstructuredUsingScheme(resource, r.config.Scheme)
+		r.deleted.Store(key, obj)
+		r.workQueue.Enqueue(obj)
+	}
+}
+
+func (r *resourceSyncer) runIfWaitForCacheSync(defaultReturn any, run func() any) any {
```
Comment (skitt) [inline pkg/syncer/resource_syncer.go]: I understand the construction, but I found “run if wait for cache sync” weird when reviewing this. `waitForCacheSyncThenRun` or `runIfCacheSynced` seem more accurate to me.

**Review by skitt — CHANGES_REQUESTED** [review]

File: pkg/syncer/resource_syncer.go (hunk 1 above)
Comment (tpantelis) [reply pkg/syncer/resource_syncer.go]: yeah it shouldn't happen  - "unknown" isn't actually an enum so perhaps "" is more appropriate but either way...I changed it back.

**Review by skitt — APPROVED** [review]

**Pull request merged**